src/universe/UniverseInferenceVisitor.java

```diff
@@ -55,101 +45,104 @@
 
     private final boolean checkOaM;
     private final boolean checkStrictPurity;
-    private final boolean allowLost;
-    private final boolean warn_staticpeer;
-
-    private final UniverseAnnotatedTypeFactory univATF;
 
-    /*
-     * We continue to use BaseTypeChecker as first parameter type, because this
-     * class is instantiated by both the UniverseInferenceChecker and the InferenceChecker.
-     */
     public UniverseInferenceVisitor(UniverseInferenceChecker checker, InferenceChecker ichecker, BaseAnnotatedTypeFactory factory, boolean infer) {
         super(checker, ichecker, factory, infer);
 
-        this.allowLost = checker.getLintOption("allowLost", false);
-        this.checkOaM = checker.getLintOption("checkOaM", false);
-        this.checkStrictPurity = checker.getLintOption("checkStrictPurity", false);
-
-        String warn = checker.getOption("warn", "");
-        warn_staticpeer = warn.contains("staticpeer");
-
-        this.univATF = (UniverseAnnotatedTypeFactory) InferenceMain.getInstance().getRealTypeFactory();
-        // System.out.println("univATF in constructor is: " + this.univATF);
-
+        checkOaM = checker.getLintOption("checkOaM", false);
+        checkStrictPurity = checker.getLintOption("checkStrictPurity", false);
     }
 
     /**
      * The type validator to ensure correct usage of ownership modifiers.
      */
     @Override
     protected UniverseInferenceValidator createTypeValidator() {
-        // System.out.println("UniverseInferenceValidator constructor is called!");
-        //System.out.println("univATF is: " + this.univATF);
-        // Overriding version of createTypeValidator() needs a parameter which is provided after the call to this method according to super constructor.
-        // But univATF is not set, so UniverseInferenceValidator's atypeFactory field is null, and causes nullpointer exception.
-        // Fix is: get realtypefactory as soon as possible, and don't wait till univATF is provided.
-
-        // I think the following line is incorrect. We need
-        // InferenceAnnotatedTypeFactory
-        // rather than UniverseAnnotatedTypeFactory.
-        /*
-        UniverseAnnotatedTypeFactory univATFParameter = (UniverseAnnotatedTypeFactory) InferenceMain.getInstance().getRealTypeFactory();
-        return new UniverseInferenceValidator(checker, this, univATFParameter);*/
-        return new UniverseInferenceValidator(checker, this, this.atypeFactory);
+        return new UniverseInferenceValidator(checker, this, atypeFactory);
+    }
+
+    @Override
+    public boolean isValidUse(AnnotatedDeclaredType declarationType, AnnotatedDeclaredType useType, Tree tree) {
+        return true;
+    }
+
+    /**
+     * Ignore method receiver annotations.
```

Comment:
Do we need this? The declared receiver must always be self and VP should then always produce something consistent. At least a comment would be useful.